### PR TITLE
[CWS] Add auto-suppression guards

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

Add auto-suppression guards to ensure it's enabled only if dump and/or profiles are enabled

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/config/config.go [hunk 1]
```diff
@@ -551,6 +551,16 @@ func (c *RuntimeSecurityConfig) sanitizeRuntimeSecurityConfigActivityDump() erro
 		return fmt.Errorf("activity dumps storage directory '%s' has to be the same than security profile storage directory '%s'", c.ActivityDumpLocalStorageDirectory, c.SecurityProfileDir)
 	}
 
+	if !c.ActivityDumpEnabled && c.ActivityDumpAutoSuppressionEnabled {
```
Comment (paulcacheux) [inline pkg/security/config/config.go:554]: I wonder if you really need `&& c.ActivityDumpAutoSuppressionEnabled` ?

**Review by YoannGh — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/config/config_linux.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
```
Comment (safchain) [inline pkg/security/config/config_linux.go:6]: nit: you can remove this line, the file suffix does the job

File: pkg/security/config/config_windows.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
```
Comment (safchain) [inline pkg/security/config/config_windows.go:6]: config_others.go + //go:build !linux

File: pkg/security/config/config_linux.go (hunk 2 above)
Comment (spikat) [reply pkg/security/config/config_linux.go]: Oh, I didn't know, fixed ! Thanks

File: pkg/security/config/config_windows.go (hunk 3 above)
Comment (spikat) [reply pkg/security/config/config_windows.go]: Fixed, thanks!

**Review by safchain — APPROVED** [review]

**Pull request merged**